core/src/main/kotlin/fr/sncf/osrd/stdcm/graph/STDCMPostProcessing.kt

```diff
@@ -99,70 +97,53 @@ private fun makeBlockWaypoints(
-    var lastRange = raw.ranges[raw.ranges.size - 1]
-    var current = lastRange.edge
+    var current = raw.edges.last()
     while (true) {
         orderedEdges.addFirst(current)
         if (current.previousNode == null) break
         current = current.previousNode!!.previousEdge
     }
-    firstRange = EdgeRange(orderedEdges.removeFirst(), firstRange.start, firstRange.end)
-    if (lastRange.edge !== firstRange.edge)
-        lastRange = EdgeRange(orderedEdges.removeLast(), lastRange.start, lastRange.end)
-    val res = ArrayList<EdgeRange<STDCMEdge, STDCMEdge>>()
-    res.add(firstRange)
-    for (edge in orderedEdges) res.add(EdgeRange(edge, Offset(0.meters), edge.length))
-    if (firstRange.edge !== lastRange.edge) res.add(lastRange)
-    return res
+    return orderedEdges.toList()
 }
 
 /** Computes the departure time, made of the sum of all delays added over the path */
-fun computeDepartureTime(ranges: List<EdgeRange<STDCMEdge, STDCMEdge>>, startTime: Double): Double {
+fun computeDepartureTime(ranges: List<STDCMEdge>, startTime: Double): Double {
     var mutStartTime = startTime
-    for (range in ranges) mutStartTime += range.edge.addedDelay
+    for (range in ranges) mutStartTime += range.addedDelay
     return mutStartTime
 }
 
 /** Builds the list of stops from the ranges */
-private fun makeStops(ranges: List<EdgeRange<STDCMEdge, STDCMEdge>>): List<TrainStop> {
+private fun makeStops(ranges: List<STDCMEdge>): List<TrainStop> {
     val res = ArrayList<TrainStop>()
     var offset = 0.meters
     for (range in ranges) {
-        val prevNode = range.edge.previousNode
-        if (
-            (prevNode != null) &&
-                (prevNode.stopDuration!! >= 0) &&
-                (range.start == Offset<STDCMEdge>(0.meters))
-        )
+        val prevNode = range.previousNode
+        if ((prevNode != null) && (prevNode.stopDuration!! >= 0))
             res.add(TrainStop(offset.meters, prevNode.stopDuration))
-        offset += range.end - range.start
+        offset += range.length.distance
     }
     return res
 }
 
 /** Builds the list of block ranges, merging the ranges on the same block */
-private fun makeBlockRanges(
-    ranges: List<EdgeRange<STDCMEdge, STDCMEdge>>
-): List<PathfindingEdgeRangeId<Block>> {
+private fun makeBlockRanges(edges: List<STDCMEdge>): List<PathfindingEdgeRangeId<Block>> {
     val res = ArrayList<PathfindingEdgeRangeId<Block>>()
     var i = 0
-    while (i < ranges.size) {
-        val range = ranges[i]
-        val start = convertOffsetToBlock(range.start, range.edge.envelopeStartOffset)
-        var length = range.end - range.start
-        while (i + 1 < ranges.size) {
-            val nextRange = ranges[i + 1]
-            if (range.edge.block != nextRange.edge.block) break
-            length += nextRange.end - nextRange.start
+    while (i < edges.size) {
+        val edge = edges[i]
+        val start = Offset<Block>(edge.envelopeStartOffset.distance)
```

Comment:
It's already an `Offset<Block>`

```suggestion
        val start = edge.envelopeStartOffset
```

Reply:
done